tests/orders/types/trailing/test_trailing_stop_order.py

```diff
@@ -13,4 +13,118 @@
+        get_exchange_symbol_data(trailing_stop_order.symbol).prices_manager
+    prices_manager.set_mark_price(new_trailing_price)
+    prices_manager.mark_price_set_time = trailing_stop_order.timestamp
+
+    # move trailing price
+    price_events_manager.handle_recent_trades(
+        [random_recent_trade(price=new_trailing_price,
+                             timestamp=trailing_stop_order.timestamp)])
+    await wait_asyncio_next_cycle()
+    assert not trailing_stop_order.is_filled()
+
+    # test fill stop loss with new order price reference
+    price_events_manager.handle_recent_trades(
+        [random_recent_trade(price=get_price_percent(new_trailing_price, trailing_stop_order.trailing_percent),
+                             timestamp=trailing_stop_order.timestamp)])
+    await wait_asyncio_next_cycle()
+    assert trailing_stop_order.is_filled()
+
+
+async def test_trailing_stop_with_new_old_price(trailing_stop_order):
+    trailing_stop_order, order_price, price_events_manager = await initialize_trailing_stop(trailing_stop_order)
+    await trailing_stop_order.set_trailing_percent(5)
+    new_trailing_price = random_price(min_value=order_price + 1)
+
+    # set mark price
+    prices_manager = trailing_stop_order.exchange_manager.exchange_symbols_data.\
+        get_exchange_symbol_data(trailing_stop_order.symbol).prices_manager
+    prices_manager.set_mark_price(new_trailing_price)
+    prices_manager.mark_price_set_time = trailing_stop_order.timestamp
+
+    # move trailing price
+    price_events_manager.handle_recent_trades(
+        [random_recent_trade(price=new_trailing_price,
+                             timestamp=trailing_stop_order.timestamp)])
+    await wait_asyncio_next_cycle()
+    assert not trailing_stop_order.is_filled()
+
+    # test fill stop loss on old price
+    price_events_manager.handle_recent_trades(
+        [random_recent_trade(price=get_price_percent(order_price, trailing_stop_order.trailing_percent) + 1,
+                             timestamp=trailing_stop_order.timestamp)])
+    await wait_asyncio_next_cycle()
+    assert trailing_stop_order.is_filled()
+
+
+async def initialize_trailing_stop(order) -> Tuple[TrailingStopOrder, float, PriceEventsManager]:
+    order_price = random_price()
+    order.update(
+        price=order_price,
+        quantity=random_quantity(),
+        symbol=DEFAULT_SYMBOL_ORDER,
+        order_type=TradeOrderType.TRAILING_STOP,
+    )
+    order.exchange_manager.is_backtesting = True  # force update_order_status
+    await order.initialize()
+    price_events_manager = order.exchange_manager.exchange_symbols_data.get_exchange_symbol_data(
+        DEFAULT_SYMBOL_ORDER).price_events_manager
+    order.exchange_manager.exchange_personal_data.orders_manager.upsert_order_instance(order)
+    return order, order_price, price_events_manager
+
+
+def get_price_percent(price, percent):
+    return price * (1 - percent / 100)
```

Comment:
👍 
Should we also test other side's trailing stops ?

Reply:
Yes we should :+1:

Reply:
Added one named `test_trailing_stop_with_new_price_inversed` :wink: